src/aap_eda/api/views/webhook.py

```diff
@@ -203,3 +208,36 @@ def partial_update(self, request, *args, **kwargs):
             serializers.WebhookOutSerializer(webhook).data,
             status=status.HTTP_200_OK,
         )
+
+    @extend_schema(
+        description="List all activations for the event stream",
+        request=None,
+        responses={
+            status.HTTP_200_OK: serializers.ActivationListSerializer(
+                many=True
+            ),
+        },
+        parameters=[
+            OpenApiParameter(
+                name="id",
+                type=int,
+                location=OpenApiParameter.PATH,
+                description="A unique integer value identifying this event stream.",  # noqa: E501
+            )
+        ],
+    )
+    @action(
+        detail=False,
+        queryset=models.Activation.objects.order_by("id"),
+        filterset_class=filters.ActivationFilter,
+        url_path="(?P<id>[^/.]+)/activations",
+    )
+    def activations(self, request, id):
+        webhook = get_object_or_404(models.Webhook, id=id)
```

Comment:
`get_object_or_404` function bypasses permission check by default, I think instead we should use `self.get_object()` or change the queryset to `models.Webhook.access_qs(request.user)` or use an approach similar to [this](https://github.com/ansible/eda-server/blob/1ccc5e415696ef5b1072e3cb9265dcf375d00c5b/src/aap_eda/api/views/activation.py#L182C1-L189C14).

Reply:
Fixed